worlds/oot_soh/__init__.py

```diff
@@ -150,6 +151,44 @@ def generate_early(self) -> None:
             self.vanilla_progressive_skulltula_count = self.passthrough["vanilla_progressive_skulltula_count"]
             self.randomized_progressive_skulltula_count = self.passthrough["randomized_progressive_skulltula_count"]
 
+        # Figure out Keyring Situation
+        key_ring_options: list = [self.options.gerudo_fortress_key_ring, self.options.forest_temple_key_ring, self.options.fire_temple_key_ring, self.options.water_temple_key_ring, self.options.spirit_temple_key_ring, self.options.shadow_temple_key_ring, self.options.bottom_of_the_well_key_ring, self.options.gerudo_training_ground_key_ring, self.options.ganons_castle_key_ring]
+
+        if self.options.key_rings == "off":
+            for index in range(len(key_ring_options)):
+                    key_ring_options[index].value = False
```

Comment:
Can't this just get replaced by the following?

```suggestion
            for option in key_ring_options:
                    option.value = False
```

Reply:
Decided to change this logic. Still implemented the change, just did it cleaner by getting rid of one of these loops..